### ISSUE TITLE
Error when Build-Module params are specified but no Build.psd1 present

### ISSUE THREAD

**Issue by gaelcolas** [PoshCode/ModuleBuilder#44]
When invoking the command Build-Module with required Parameters like this:
`Build-Module -Path .\Source\ModuleBuilder.psd1 -OutputDirectory ..\ -VersionedOutputDirectory`
but the `Build.psd1` file is absent, we get the error below.
The only way to work around it is to create a `Build.psd1` with an empty hashtable.

I think we should make that file optional when required parameters are provided, and clearer error when not.

```
> build-module -Path .\Source\ModuleBuilder.psd1 -OutputDirectory ..\ -VersionedOutputDirectory
Import-Metadata : Can't find file C:\src\ModuleBuilder\Source\[Bb]uild.psd1
At C:\src\ModuleBuilder\output\ModuleBuilder\1.0.0\ModuleBuilder.psm1:83 char:18
+ ... BuildInfo = Import-Metadata -Path (Join-Path $ModuleSource [Bb]uild.p ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (:) [Import-Metadata], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Metadata\Import-Metadata,Import-Metadata

InitializeBuild : You cannot call a method on a null-valued expression.
At C:\src\ModuleBuilder\output\ModuleBuilder\1.0.0\ModuleBuilder.psm1:610 char:27
+             $ModuleInfo = InitializeBuild $SourcePath
+                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [InitializeBuild], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull,InitializeBuild
```

**Comment by Jaykul** [comment]
Ultimately, build.psd1 is basically our project file -- we're even calculating relative paths relative to the build.psd1 -- if it was called build.psproj would you still be trying to get rid of it? I suppose you would.

I don't know. 

One problem with this is: what is `-OutputDirectory ..\` relative to, if there's no `build.psd1` file?

**Comment by Jaykul** [comment]
So, now that all the paths are relative to the module.psd1 instead of the build.psd1 ...

**Comment by gaelcolas** [comment]
I'm waiting for the current version to be released before tackling this one, as I need it for my current project.
/cc @Jaykul

**Comment by Jaykul** [comment]
Sorry that took so long.

**Comment by johlju** [comment]
@gaelcolas This issue should be resolved once the PR #90 is merged?
